src/libraries/Microsoft.PowerFx.Core/Public/Types/UserProvidedTypeDefinitions/FormulaTypeToSchemaHelper.cs

```diff
@@ -0,0 +1,153 @@
+                        RecordType recordType => recordType.ToTable(),
+
+                        // Add support for table of primitives here in the future
+                        _ => FormulaType.BindingError,
+                    };
+                }
+                
+                return actualType;
+            }
+
+            if (typeName != RecordTypeName.Name)
+            {
+                return FormulaType.BindingError;
+            }
+
+            if (schema.Fields == null || !schema.Fields.Any())
+            {
+                return FormulaType.BindingError;
+            }
+
+            var result = new UserDefinedRecordType(schema, definedTypeSymbols);
+            return schema.Type.IsTable ? result.ToTable() : result;
+        }
+
+        private static SchemaTypeName RecordTypeName => new () { Name = "Record", IsTable = false };
+
+        private static SchemaTypeName TableTypeName => new () { Name = "Record", IsTable = true };
+
+        private static FormulaTypeSchema ToSchema(FormulaType type, DefinedTypeSymbolTable definedTypeSymbols, int maxDepth)
+        {
+            if (maxDepth < 0)
+            {
+                throw new InvalidOperationException("Max depth exceeded when converting type to schema definition");
+            }
+
+            if (TryLookupTypeName(type, definedTypeSymbols, out var typeName))
+            {
+                return new FormulaTypeSchema()
+                {
+                    Type = new SchemaTypeName() { Name = typeName }
+                };
+            }
+
+            // Possible that the record type is defined for a TableType variant
+            if (type is TableType tableType && TryLookupTypeName(tableType.ToRecord(), definedTypeSymbols, out typeName))
+            {                
+                return new FormulaTypeSchema()
+                {
+                    Type = new SchemaTypeName() { Name = typeName, IsTable = true }
+                };
+            }
+
+            if (type is not AggregateType aggregateType)
+            {
+                throw new InvalidOperationException($"Conversion to schema definition not supported for type {type}");
+            }
+
+            var children = GetChildren(aggregateType, definedTypeSymbols, maxDepth - 1);
+
+            if (aggregateType is RecordType)
+            {
+                return new FormulaTypeSchema()
+                {
+                    Type = RecordTypeName,
+                    Fields = children
+                };
+            }                
+                
+            return new FormulaTypeSchema()
+            {
+                Type = TableTypeName,
+                Fields = children
+            };
+        }
+
+        private static bool TryLookupType(string typeName, DefinedTypeSymbolTable definedTypeSymbols, out FormulaType type)
+        {
+            var lookupOrder = new List<TypeSymbolTable>() { definedTypeSymbols, PrimitiveTypesSymbolTable.Instance };
```

Comment:
>definedTypeSymbols [](http://example.com/codeflow?start=60&length=18)

Is the precedence given to symboltable than primitive types? Is that correct?

Reply:
Yes, that's correct.